src/corecel/sys/ScopedSignalHandler.cc

```diff
@@ -26,7 +26,12 @@ sig_atomic_t volatile g_celer_signal_bits_ = 0;
 //! Set the bit corresponding to a signal
 extern "C" void celer_set_signal(int signal)
 {
+#ifndef _WIN32
+    // Windows exception handling settings do not allow throwing from a C
+    // function. It's not safe to do anyway if called from the signal handler,
+    // so expect this to terminate the code if the signal is invalid.
```

Comment:
As a consequence, shouldn't we use a non-throwing assert here both for Linux and Windows?

Reply:
I think calling `CELER_ASSERT` and invoking UB is still cleaner than
```c++
if (CELERITAS_DEBUG && !(signal >= 0 && signal < static_cast<int>(sizeof(int) * 8 - 1)))
{
    std::cerr << "Got out-of-range signal at " << __FILE__ << ":" << __LINE__  << '\n';
    std::abort();
}
```
for this very edge case that shouldn't even throw...

Reply:
Since this is an edgey/unlikely case, wouldn't:
```
assert(signal >= 0 && signal < static_cast<int>(sizeof(int) * 8 - 1));
```
be sufficient?

> invoking UB is still cleaner than

Technically, that "could" be (since it is undefined :) ): not-throw the exception and ignore it completely.

Reply:
Yeah good call. If we're using C signals, it's probably better to use C assertions.